examples/validator.py

```diff
@@ -0,0 +1,75 @@
+"""A basic example showing how to use field validators to validate the input data.
+
+Note: All validators are defined in the class because updates to the validators will
+affect the output of the model.
+"""
+import os
+from typing import Union
+
+from pydantic import (
+    BaseModel,
+    Field,
+    ValidationError,
+    field_validator,
+)
+
+from mirascope import OpenAICallParams, OpenAIChat, Prompt
+
+os.environ["OPENAI_API_KEY"] = "YOUR_API_KEY"
+
+chat = OpenAIChat()
+
+topic = "baking"
+
+
+class QuestionPrompt(Prompt):
+    """
+    SYSTEM:
+    You are an expert in {topic}.
+
+    USER:
+    {question}
+    """
+
+    topic: str = topic
+    question: str = Field(..., description="The user's question.")
+
+    call_params = OpenAICallParams(model="gpt-3.5-turbo", temperature=0.1)
+
+    @field_validator("question", mode="after")
+    @classmethod
+    def validate_question(cls, question: str) -> str:
+        class Text(BaseModel):
```

Comment:
marge into `ValidationInfo` and do single extract

Reply:
Follow-up:

What if we showed hitting a "fine-tuned" local model for validation before sending the request to API to scrub it of PII